File: PrivateCloudDisk-win/PrivateCloudDisk/Models/UserModels.cs
using System.Text.Json.Serialization;

namespace PrivateCloudDisk.Models;

/// <summary>
/// 用户信息 (对应后端 UserVO)
/// </summary>
public class UserProfile
{
    [JsonPropertyName("user_id")]
    public string UserId { get; set; } = string.Empty;

    [JsonPropertyName("account")]
    public string Account { get; set; } = string.Empty;

    [JsonPropertyName("user_name")]
    public string UserName { get; set; } = string.Empty;

    [JsonPropertyName("phone_number")]
    public string? PhoneNumber { get; set; }

    [JsonPropertyName("email")]
    public string? Email { get; set; }

    [JsonPropertyName("avatar_url")]
    public string? AvatarUrl { get; set; }

    public string DisplayName => string.IsNullOrWhiteSpace(UserName) ? Account : UserName;
}

/// <summary>
/// 登录请求 (对应后端 LoginRequest)
/// </summary>
public class LoginRequest
{
    /// <summary>账号 (必填)</summary>
    [JsonPropertyName("account")]
    public string Account { get; set; } = string.Empty;

    /// <summary>手机号 (可选)</summary>
    [JsonPropertyName("phone_number")]
    public string? PhoneNumber { get; set; }

    /// <summary>密码 (必填)</summary>
    [JsonPropertyName("password")]
    public string Password { get; set; } = string.Empty;
}

/// <summary>
/// 登录响应 (对应后端 LoginResponse)
/// </summary>
public class LoginResponse
{
    [JsonPropertyName("token")]
    public string Token { get; set; } = string.Empty;

    [JsonPropertyName("user_id")]
    public string UserId { get; set; } = string.Empty;

    [JsonPropertyName("user_name")]
    public string UserName { get; set; } = string.Empty;
}

/// <summary>
/// 注册请求 (对应后端 RegisterRequest)
/// </summary>
public class RegisterRequest
{
    [JsonPropertyName("account")]
    public string Account { get; set; } = string.Empty;

    [JsonPropertyName("user_name")]
    public string UserName { get; set; } = string.Empty;

    [JsonPropertyName("password")]
    public string Password { get; set; } = string.Empty;

    [JsonPropertyName("phone_number")]
    public string? PhoneNumber { get; set; }

    [JsonPropertyName("email")]
    public string? Email { get; set; }
}

/// <summary>
/// 更新用户信息请求 (对应后端 UpdateUserInfoRequest)
/// </summary>
public class UpdateUserInfoRequest
{
    [JsonPropertyName("new_name")]
    public string? NewName { get; set; }

    [JsonPropertyName("new_phone_number")]
    public string? NewPhoneNumber { get; set; }

    [JsonPropertyName("new_email")]
    public string? NewEmail { get; set; }
}

/// <summary>
/// 修改密码请求 (对应后端 ChangePasswordRequest)
/// </summary>
public class ChangePasswordRequest
{
    [JsonPropertyName("user_password")]
    public string UserPassword { get; set; } = string.Empty;

    [JsonPropertyName("new_password")]
    public string NewPassword { get; set; } = string.Empty;
}

File: PrivateCloudDisk-android/app/src/main/java/com/privateclouddisk/android/data/model/UserModels.java
package com.privateclouddisk.android.data.model;

import com.google.gson.annotations.SerializedName;

/**
 * 用户信息（对应后端 UserVO）
 */
public class UserProfile {

    @SerializedName("user_id")
    private String userId;

    @SerializedName("account")
    private String account;

    @SerializedName("user_name")
    private String userName;

    @SerializedName("phone_number")
    private String phoneNumber;

    @SerializedName("email")
    private String email;

    @SerializedName("avatar_url")
    private String avatarUrl;

    // ── Getters & Setters ──
    public String getUserId() { return userId; }
    public void setUserId(String userId) { this.userId = userId; }

    public String getAccount() { return account; }
    public void setAccount(String account) { this.account = account; }

    public String getUserName() { return userName; }
    public void setUserName(String userName) { this.userName = userName; }

    public String getPhoneNumber() { return phoneNumber; }
    public void setPhoneNumber(String phoneNumber) { this.phoneNumber = phoneNumber; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }

    public String getAvatarUrl() { return avatarUrl; }
    public void setAvatarUrl(String avatarUrl) { this.avatarUrl = avatarUrl; }

    /** 显示名称：优先用 userName，否则用 account */
    public String getDisplayName() {
        return (userName != null && !userName.isEmpty()) ? userName : account;
    }
}

/**
 * 登录请求
 */
class LoginRequest {

    @SerializedName("account")
    private String account;

    @SerializedName("phone_number")
    private String phoneNumber;

    @SerializedName("password")
    private String password;

    public LoginRequest(String account, String password) {
        this.account = account;
        this.password = password;
    }

    public String getAccount() { return account; }
    public String getPassword() { return password; }
    public String getPhoneNumber() { return phoneNumber; }
    public void setPhoneNumber(String phoneNumber) { this.phoneNumber = phoneNumber; }
}

/**
 * 登录响应
 */
class LoginResponse {

    @SerializedName("token")
    private String token;

    @SerializedName("user_id")
    private String userId;

    @SerializedName("user_name")
    private String userName;

    public String getToken() { return token; }
    public String getUserId() { return userId; }
    public String getUserName() { return userName; }
}

/**
 * 注册请求
 */
class RegisterRequest {

    @SerializedName("account")
    private String account;

    @SerializedName("user_name")
    private String userName;

    @SerializedName("password")
    private String password;

    public RegisterRequest(String account, String userName, String password) {
        this.account = account;
        this.userName = userName;
        this.password = password;
    }

    public String getAccount() { return account; }
    public String getUserName() { return userName; }
    public String getPassword() { return password; }
}

/**
 * Token 验证请求
 */
class TokenValidationRequest {

    @SerializedName("token")
    private String token;

    public TokenValidationRequest(String token) {
        this.token = token;
    }
}

/**
 * Token 刷新请求
 */
class TokenRefreshRequest {

    @SerializedName("refresh_token")
    private String refreshToken;

    public TokenRefreshRequest(String refreshToken) {
        this.refreshToken = refreshToken;
    }
}

/**
 * Token 刷新响应
 */
class TokenRefreshResponse {

    @SerializedName("token")
    private String token;

    @SerializedName("refresh_token")
    private String refreshToken;

    public String getToken() { return token; }
    public String getRefreshToken() { return refreshToken; }
}